packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -99,6 +99,32 @@ contract MetaTransactionWallet is
     emit EIP712DomainSeparatorSet(eip712DomainSeparator);
   }
 
+  /**
+   * @notice Returns the struct hash of the MetaTransaction
+   * @param destination The address to which the meta-transaction is to be sent.
+   * @param value The CELO value to be sent with the meta-transaction.
+   * @param data The data to be sent with the meta-transaction.
+   * @param _nonce The nonce for this meta-transaction local to this wallet.
+   * @return The digest of the provided meta-transaction.
+   */
+  function _getMetaTransactionStructHash(
+    address destination,
+    uint256 value,
+    bytes memory data,
+    uint256 _nonce
+  ) internal view returns (bytes32) {
+    return
+      keccak256(
+        abi.encode(
+          EIP712_EXECUTE_META_TRANSACTION_TYPEHASH,
+          destination,
+          value,
+          keccak256(data),
```

Comment:
everything looks good to me. There was a side conversation (@nambrot was there) in slack that mentioned that this keccak256 is extra and useless. Getting rid of it might complicate things on your end, so I don't think it's a big deal, but if it's easy, we don't need this.